### PR TITLE
Changing visibility of types required for api

### PR DESCRIPTION
**Pull request by mettke**
* LocalRoomNameParseError:
https://docs.rs/screeps-game-api/0.4.0/screeps/struct.LocalRoomName.html
* Bodypart, MoveToOptions
https://docs.rs/screeps-game-api/0.4.0/screeps/objects/struct.Creep.html
* PortalDestination
https://docs.rs/screeps-game-api/0.4.0/screeps/objects/struct.StructurePortal.html

### CODE REVIEW

File: screeps-game-api/src/objects/impls/mod.rs [hunk 1]
```diff
@@ -30,6 +30,8 @@ pub use self::{
         HarvestEvent, HealEvent, HealType, LookResult, ObjectDestroyedEvent, Path,
         PositionedLookResult, RepairEvent, ReserveControllerEvent, Step, UpgradeControllerEvent,
     },
+    creep::{Bodypart, MoveToOptions},
```
Comment (ASalvail) [inline screeps-game-api/src/objects/impls/mod.rs]: Those should be accessible through their own sub-modules, aren't they?

**Comment by mettke** [comment]
Sry, I thought linking the docs would explain the problem sufficiently. Lets elaborate a bit.

In my experience with the rustdoc, if the documentation does not show a type it is not visible to the outside. This is backed up by the fact that `mod positions;` and `mod impls;` are both private modules. Thus, even thou the types themselves are marked as public, they will only be available to submodules and not a third party.

**Comment by daboross** [comment]
Thanks!

I think this is definitely my fault for having a confusing module structure which allows for public-marked types which don't end up actually public.

**Comment by ASalvail** [comment]
Sure! I'm waiting for the changes to be passed through `rustfmt` to approve.

**Comment by mettke** [comment]
@ASalvail I see - sry guess I was a bit confused or something. Completely missed the point of clippy and rustfmt. Will do that this afternoon.

**Comment by daboross** [comment]
Hi!

Sorry, I completely missed that last commit - had the notification for the comment but missed that the change had been made.

@ASalvail Hope it's alright if I just merge this - the formatting seems to be fixed up.

And thanks again @mettke!

**Comment by ASalvail** [comment]
I also missed the commit, so I'm good. Thanks @daboross 
And thanks @mettke for this PR (and your patience!)

**Pull request merged**